src/mol-model/structure/structure/util/lookup3d.ts

```diff
@@ -39,6 +41,25 @@ export namespace StructureResult {
         out.count = result.count;
         return out;
     }
+
+    // sort in ascending order based on squaredDistances
+    export function sort(result: StructureResult) {
+        const { indices, squaredDistances, units, count } = result;
+        if (indices.length > count) {
+            // clear arrays before doing sorting if necessary
+            indices.length = count;
+            squaredDistances.length = count;
+            units.length = count;
+        }
+        for (let i = 1, c = result.count; i < c; i++) {
+            if (arrayLess(squaredDistances, i - 1, i) > 0) {
+                arraySwap(squaredDistances, i - 1, i);
+                arraySwap(indices, i - 1, i);
+                arraySwap(units, i - 1, i);
+                i = Math.max(0, i - 2);
+            }
+        }
```

Comment:
What kind of sort is this? What you are looking for is a structure called heap (binary could be enough, but perhaps fibonacci better) to keep the list of the nearest points.

Reply:
I have found this project, https://github.com/gwtw/ts-fibonacci-heap do you think this could work for our use case?

Reply:
It might and since it's MIT license we can also clone the code into the `mol-util` package (with credits in the file header) instead of relying on this as a dependency.

Tho I am a bit worried about their use of the iterator object as that is quite slow to do the allocations. But that could be improved.

Reply:
Also, we need to be careful about the size of the heap as this implementation has O(n) complexity for this operation.